tests/python_package_test/test_engine.py

```diff
@@ -2278,6 +2311,99 @@ def test_multiple_feval_train():
     assert 'decreasing_metric' in evals_result['valid_0']
 
 
+def test_objective_callable_train_binary_classification():
+    X, y = load_breast_cancer(return_X_y=True)
+    params = {
+        'verbose': -1,
+        'objective': logloss_obj,
+        'learning_rate': 0.01
+    }
+    train_dataset = lgb.Dataset(X, y)
+    booster = lgb.train(
+        params=params,
+        train_set=train_dataset,
+        num_boost_round=100,
+        feval=logloss_metric
```

Comment:
as mentioned in https://github.com/microsoft/LightGBM/pull/5052#pullrequestreview-913851568, please remove any changes from this PR related to custom *metric* and keep it focused only on custom objective functions

Reply:
copy that